### PR TITLE
Fix initializer indentation for 'new()' expression

### PR DESCRIPTION
**Pull request by alrz**
Fixes https://github.com/dotnet/roslyn/issues/47442

### CODE REVIEW

**Comment by alrz** [comment]
@jcouv @CyrusNajmabadi

**Comment by alrz** [comment]
tagging @sharwell since he did the same fix for property patterns. Is there any specific scenario to test here?

**Comment by sharwell** [comment]
💭 Should we apply `IndentBlockOption.IndentIfConditionOfAnchorToken` here, the same way we did for property patterns in https://github.com/dotnet/roslyn/commit/b0af62ef354d04e209e89a077d5582b9d757b8a7?

**Review by sharwell — APPROVED** [review]

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
>  I'm thinking we leave it for a separate pull request because this form for new() is not typically seen in conditions.

That seems very reasonable to me. I also could not think of any cases where that would be needed.

File: src/Workspaces/CSharpTest/Formatting/FormattingTests.cs [hunk 1]
```diff
@@ -9603,5 +9603,43 @@ void M([My][My] int x)
...
+        => new C()
+    {
+        Name = name
+    };
+    public static C Create2(string name)
+        => new()
+    {
+        Name = name
+    };
+}";
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharpTest/Formatting/FormattingTests.cs:9625]: How are these two cases different?  (Asking as they look identical to be)

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Workspaces/CSharpTest/Formatting/FormattingTests.cs (hunk 1 above)
Comment (alrz) [reply src/Workspaces/CSharpTest/Formatting/FormattingTests.cs:9625]: just to make sure we match `new C()` formatting, however that may be.

**Comment by alrz** [comment]
I did see IndentBlockOption.IndentIfConditionOfAnchorToken but decided to follow obj cretion behaviour.

btw what is an "anchor token"?

**Comment by CyrusNajmabadi** [comment]
>btw what is an "anchor token"?

A way to attach one nodes movement to another entity. Effectively, if the thing you are achored to changes indentation, your node should change indentation the same amount.

File: src/Workspaces/CSharpTest/Formatting/FormattingTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/CSharpTest/Formatting/FormattingTests.cs:9625]: Ah got it. Thanks. I totally didn't see that :-)

**Comment by CyrusNajmabadi** [comment]
@alrz doing anything else here, or do you want me to merge?  Thanks!

**Comment by alrz** [comment]
This is ready to go ✔️

**Comment by jcouv** [comment]
Thanks @alrz! :heart:

**Pull request merged**